pmtct_missed_sets.py

```diff
@@ -0,0 +1,133 @@
+    if messageset_id not in schedules:
+        headers = {
+            'Authorization': "Token " + token,
+            'Content-Type': "application/json"
+        }
+        r = session.get('%smessageset/%s' % (url, messageset_id),
+                        headers=headers)
+        r.raise_for_status()
+
+        messageset = r.json()
+        schedules[messageset_id] = messageset['default_schedule']
+
+    return schedules[messageset_id]
+
+
+parser = argparse.ArgumentParser(description='Subscribe PMTCT users to a'
+                                 'important message set.')
+parser.add_argument('--sbm-url', required=True,
+                    help='The url for the Stage Based Messaging service.')
+parser.add_argument('--sbm-token', required=True,
+                    help='The token for the Stage Based Messaging service.')
+parser.add_argument('--file', dest='data_file', type=argparse.FileType('r'),
+                    help='Name of file containing the list of identities.')
+parser.add_argument('--data', help='List of identities. One per line.')
+
+args = parser.parse_args()
+sbm_url = args.sbm_url
+sbm_token = args.sbm_token
+
+if args.data_file:
+    identity_list = args.data_file.readlines()
+elif args.data:
+    identity_list = args.data.split("\n")
+
+for item in identity_list:
+    identity = json.loads(item)
+
+    old_set_id = identity['current_messageset_id']
+    new_set_id = identity['expected_messageset_id']
+    old_seq = identity['current_sequence_number']
+    new_seq = identity['expected_sequence_number']
+
+    start = False
+
+    # if this is a PMTCT message set
+    if old_set_id in set_details.keys():
+
+        # loop through the message set
+        for key, data in set_details.items():
+
+            # if we reach the first set of sub or if we have started
+            if key == old_set_id or start:
+                start = True
+                start_seq = None
+
+                # we loop through important messages
+                for i, val in enumerate(data['seq']):
+
+                    # if our current sequence is before a important messages
+                    # and our after sequence is after a important message or
+                    # we have to move to the next set, we know we have to
+                    # subscribe and start at this sequence
+                    if old_seq < val and (new_seq >= val or key != new_set_id):
+                        start_seq = i + 1
+                        break
+
+                # create subscription with on set data['new'] on start_seq
+                if start_seq:
+                    try:
+                        messageset_schedule = get_messageset_schedule(
+                            sbm_url, sbm_token, data['new'])
+                    except requests.HTTPError as e:
+                        sys.exit("Problem retrieving the messageset: %s" %
+                                 e.response.status_code)
+                        break
```

Comment:
I think we would want to log the identity ID here, so we know where to carry on from if we have an error.